cpp_linter/rest_api/github_api.py

```diff
@@ -516,13 +584,195 @@ def _dismiss_stale_reviews(self, url: str):
                     and cast(str, review["body"]).startswith(COMMENT_MARKER)
                     and "state" in review
                     and review["state"] not in ["PENDING", "DISMISSED"]
+                    and (
+                        ignored_reviews is not None
+                        and review["node_id"] not in ignored_reviews
```

Comment:
If `ignored_reviews` is not given, then dismiss all reviews. If `ignored_reviews` is given, then only dismiss the unspecified reviews.
```suggestion
                        ignored_reviews is None
                        or (
                            "node_id" in review
                            and review["node_id"] not in ignored_reviews
                        )
```

> [!tip]
> This takes advantage of python's lazy evaluation in conditional `or` statements.
> Meaning, if the left hand boolean value is True, then the right hand boolean value is ignored.
>
> In the above suggestion, the `mypy` tool (for type checking) is smart enough to understand that `ignored_reviews` is not `None` when the `or`'s right hand side is evaluated.